doc/superbuild/source/developers/python/index.rst

```diff
@@ -0,0 +1,246 @@
+
+- Cannot bind to functions which take double, or more pointer arguments. I.e., it cannot bind to `**` or `***` and so on. These have to be flattened somehow.
+- Cannot implicitly convert between a "View" container class and the underlying C type. I.e., it cannot implicitly convert ``ARKodeView`` to ``void*``.
+   This means user must explicitly convert from the "View" class by calling the ``get`` member function.
+
+We use `litgen <https://github.com/pthom/litgen>__` to generate a large portion of the nanobind code.
+
+- We have ``generate.yaml`` files designate headers to generate bindings from and functions to exclude.
+- A ``generate.py`` script uses litgen to generate the bindings as a C++ header according to the ``generate.yaml``.
+- For each generated file, there is at least one hand-coded file that includes the generated header.
+
+.. note::
+
+    Litgen itself is licensed under GPLv3. This means the ``generate.py`` script is effectively governed by ``GPLv3``,
+    **but the binding code generated by the script/litgen falls only under our SUNDIALS license**.
+    Because of this, the ``generate.py`` script and litgen extensions are kept in a separate git repository as package,
+    ``sundials4py-generator``.
+
+
+Structure
+---------
+
+sundials4py code lives in ``bindings/sundials4py``. The main python module and all of its submodules are defined in ``sundials4py.cpp``.
+sundials4py consists of 5 modules. Below we list how each one maps to the directory layout:
+
+- **sundials.arkode**:
+  - Implements bindings for all of ARKODE.
+  - Source directory: ``arkode/``
+
+- **sundials.cvodes**:
+  - Provides bindings for all of CVODES.
+  - Source directory: ``cvodes/``
+
+- **sundials.idas**:
+  - Contains bindings for all of IDAS.
+  - Source directory: ``idas/``
+
+- **sundials.kinsol**:
+  - Facilitates bindings for all of KINSOL.
+  - Source directory: ``kinsol/``.
+
+- **sundials.core**:
+  - All SUNDIALS shared classes/modules and implementations.
+  - Source directories:
+    - `nvector/`
+    - `sunadaptcontroller/`
+    - `sunadjointcheckpointscheme/`
+    - `sunlinsol/`
+    - `sunmatrix/`
+    - `sunmemory/`
+    - `sunnonlinsol/`
+    - `sundials/`
+    - `sundomeigest/`
+
+
+Development
+-----------
+
+The recommended method for development is to use a typical Python development workflow with ``pip`` rather than invoking CMake directly.
+
+.. code-block:: shell
+
+   cd sundials_root_directory
+   python -m venv .venv  # create python virtual environment
+   . .venv/bin/activate  # activate the python virtual environment
+   pip install scikit-build-core[pyproject] hatchling nanobind # this is a prerequisite for the next step
+   MAKEFLAGS="-j$(nproc)" pip install --no-build-isolation -Ceditable.rebuild=true -ve .[dev] # install sundials4py into the virtual environment
+
+The last ``pip install`` command will allow automatic incremental builds. It will invoke the SUNDIALS `CMake` build system with the
```

Comment:
```suggestion
The last ``pip install`` command will allow automatic incremental builds. It will invoke the SUNDIALS ``CMake`` build system with the
```